recipes/hipace/meta.yaml

```
@@ -0,0 +1,93 @@
+{% set name = "hipace" %}
+{% set version = "23.03" %}
+{% set build = 0 %}
+{% set sha256 = "f32104af1e7ca33b7ef8a3fd7964ee62db13a68a00af305d4ace5d0e86e2efb2" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/Hi-PACE/hipace/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: {{ build }}
+  run_exports:
+    # releases are not (yet) compatible
+    - hipace {{ version }}
+
+requirements:
+  build:
+    - python                                 # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - ccache        # [unix]
+    - clang         # [win]
+    - cmake >=3.20.0
+    - lld           # [win]
+    # TODO: add libxml2 dep in lld-feedstock
+    - libxml2       # [win]
+    - make          # [unix]
+    - ninja         # [win]
+    - libgomp       # [linux]
+    - llvm-openmp   # [osx or win]
+  host:
+    - fftw
+    - openpmd-api
+    - pkgconfig
+    # ctest
+    - matplotlib-base  # [build_platform == target_platform]
+    - numpy            # [build_platform == target_platform]
+    - openpmd-viewer   # [build_platform == target_platform]
+    - pandas           # [build_platform == target_platform]
+    - pytest           # [build_platform == target_platform]
+    - python           # [build_platform == target_platform]
+    - scipy            # [build_platform == target_platform]
```

```suggestion
    - pkgconfig
  host:
    - fftw
    - openpmd-api
    - numpy
    - python
```
My suggestion to get a solveable environment is to try removing all of these dependencies that are only used for testing.

You can either skip running unit tests (unit tests are the upstream project's responsibility) or delay them to the test phase instead of doing them during the build phase.

Also, pkgconfig is a build dependency not a host dependency?

---

Thanks! The reason so many of the test dependencies are in the build phase is that they are used during the `ctest` phase, which is run in the build directory.

---

I try to add `build/` to the `source_files:` section to defer the `ctest` run to the test phase... let's see if that can work.